platform/CriticalSectionLock.h

```diff
@@ -56,20 +67,38 @@ class CriticalSectionLock {
     }
 
     /** Mark the start of a critical section
-     *     
+     *
      */
+    MBED_DEPRECATED_SINCE("mbed-os-5.8",
+            "This function is inconsistent with RAII and is being removed in the future.")
```

Comment:
The deprecated note should have the name syntax of what replaces it. Something like
```" This function is inconsistent with RAII and is being removed in the future. For non RAII based API use CriticalSectionLock::enable();")